bolt-contracts/src/contracts/BoltRegistry.sol

```diff
@@ -92,6 +97,15 @@ contract BoltRegistry is IBoltRegistry {
             revert CooldownNotElapsed();
         }
 
+        // Remove operator from the operators array
+        for (uint256 i = 0; i < operators.length; i++) {
+            if (operators[i] == msg.sender) {
+                operators[i] = operators[operators.length - 1];
```

Comment:
Is this how you remove in solidity? Can't you just `delete operators[i]`?

Reply:
delete will just set that value to 0 instead of resizing the array

e.g. before and after:  [1, 2, 3, 4, 5] -> [1, 2, 0, 4, 5]
where we want: [1, 2, 4, 5]

Reply:
With this example, assume `i` equals `2`. Then `operators[i]` will be set to `operators[4]` which is 5, so we have the list `[1,2,5,4,5]`, then we pop so we're left with `[1,2,5,4]`. Am I missing something?

Reply:
@merklefruit has explained this right! This should be correct imo